src/main.rs

```diff
@@ -1,51 +1,78 @@
-use std::io::{Write, stdout};
+use std::{io::{Write, stdout}, fs, collections::HashMap};
 use clap::Parser;
 use crate::game::Game;
 
 pub mod game;
 
-#[derive(Parser,Debug)]
+#[derive(Parser, Debug)]
 struct Args {
     #[arg(short, default_value_t = 6)]
-    tries:u32,
+    tries: u32,
 
     #[arg(short, default_value_t = 5)]
-    word_length:usize,
+    word_length: usize,
+
+    #[arg(short,long, default_value = "english")]
+    language: String,
 }
 
 fn clear(){
     println!("{}[2J", 27 as char);
 }
 
+fn load_languages() -> HashMap<String, HashMap<String,String>>{
+    let mut lang_map:HashMap<String, HashMap<String,String>> = HashMap::new();
+    let langs = fs::read_dir("./languages").unwrap();
+    for lang in langs{
+        let file_name = lang.unwrap().file_name();
+        let lang_name = file_name.to_str().unwrap().trim_end_matches(".txt").to_string();
+        let mut map:HashMap<String,String> = HashMap::new();
+        let file = fs::read_to_string(format!("./languages/{}",file_name.to_str().unwrap()));
+        for line in file.unwrap().lines() {
+            let key = line.split(":").next().unwrap().to_string();
+            let value = line.split(":").nth(1).unwrap().to_string();
+            map.insert(key, value);
+        }
+        lang_map.insert(lang_name, map);
+    }
+    lang_map
```

Comment:
Her kunne det være en god idé at bruge et parsing library, som serde i stedet for at genopfinde hjulet selv. Koden her er ikke umiddelbart til at læse.
I stedet for at lave et mutable hashmap og inserte, kan man også collecte en iterator til et hashmap med .collect, som både er mere læseligt og hurtigere.

Reply:
Hvis du brugte et parsing library kunne du parse til et struct og dermed undgå alle de lookups du har senere, som ikke er statisk typede, som giver større risiko for fejl